crates/bevy_xr_utils/src/tracking_utils.rs

```diff
@@ -0,0 +1,314 @@
+    mut stage_query: Query<&mut Transform, (With<XRTrackedStage>, Without<XrTrackingRoot>)>,
+) {
+    let tracking_root_transform = root_query.get_single_mut();
+    match tracking_root_transform {
+        Ok(root) => {
+            for (mut transform) in &mut stage_query {
+                *transform = root.clone();
+            }
+        }
+        Err(_) => (),
+    }
+}
+
+//view
+#[derive(Component)]
+struct HeadXRSpace(XrReferenceSpace);
+
+fn update_head_transforms(
+    session: Res<OxrSession>,
+    default_ref_space: Res<XrPrimaryReferenceSpace>,
+    pipelined: Option<Res<Pipelined>>,
+    frame_state: Res<OxrFrameState>,
+    mut query: Query<(&mut Transform, &HeadXRSpace, Option<&XrReferenceSpace>)>,
+) {
+    for (mut transform, space, ref_space) in &mut query {
+        let ref_space = ref_space.unwrap_or(&default_ref_space);
+        let time = if pipelined.is_some() {
+            openxr::Time::from_nanos(
+                frame_state.predicted_display_time.as_nanos()
+                    + frame_state.predicted_display_period.as_nanos(),
+            )
+        } else {
+            frame_state.predicted_display_time
+        };
+        let space_location = session.locate_space(&space.0, ref_space, time);
+
+        if let Ok(space_location) = space_location {
+            let flags = OxrSpaceLocationFlags(space_location.location_flags);
+            if flags.pos_valid() {
+                transform.translation = space_location.pose.position.to_vec3();
+            }
+            if flags.rot_valid() {
+                transform.rotation = space_location.pose.orientation.to_quat();
+            }
+        }
+    }
+}
+
+fn update_view(
+    mut head_query: Query<&mut Transform, (With<HeadXRSpace>, Without<XRTrackedView>)>,
+    mut view_query: Query<&mut Transform, (With<XRTrackedView>, Without<HeadXRSpace>)>,
+) {
+    let head_transform = head_query.get_single_mut();
+    match head_transform {
+        Ok(root) => {
+            for (mut transform) in &mut view_query {
+                *transform = root.clone();
+            }
+        }
+        Err(_) => (),
+    }
+}
+
+//local floor
+#[derive(Component)]
+struct LocalFloor;
+//internal
+fn update_local_floor_transforms(
+    mut head_space: Query<&mut Transform, (With<HeadXRSpace>, Without<LocalFloor>)>,
+    mut local_floor: Query<&mut Transform, (With<LocalFloor>, Without<HeadXRSpace>)>,
+) {
+    let head_transform = head_space.get_single_mut();
+    match head_transform {
+        Ok(head) => {
+            let mut calc_floor = head.clone();
+            calc_floor.translation.y = 0.0;
+            //TODO: use yaw
+            let (y, x, z) = calc_floor.rotation.to_euler(EulerRot::YXZ);
+            let new_rot = Quat::from_rotation_y(y);
+            calc_floor.rotation = new_rot;
+            for (mut transform) in &mut local_floor {
+                *transform = calc_floor;
+            }
+        }
+        Err(_) => (),
+    }
+}
```

Comment:
any reason to not just use the runtime impl of this?

Reply:
local floor is not supported on all platforms, I should make it more clear this is emulated.

Reply:
maybe only use this as a fallback? also i think it is supported on every 1.1 runtime

Reply:
this was changed to direciton change the entities with the markers